Hash类

ZhangNan13 opened the pull request:
1.对文件存储及读取方法进行了修正，将路径变量添加为输入值，增加了文件存储及读取方法的可复用性。
2.新增了initsetting方法，用于创建存储git内容的文件夹，解决文件存储时无法找到路径的问题。
3.新增了测试部分内容。
@wxc1217 @RitaPeng0101

ZhangNan13 committed:
第一周任务修改

对第一周的任务进行了修改，此部分中实现了Hash值的计算以及key-value的存储。
Hash值的计算参考课上提出的问题修改了原有代码。
在存储部分，将单个key-value以文件形式存储在规定路径E:\\MyGit\\object中。存储前先寻找是否存在文件，若存在key值为文件名的文件，不会重复存储。
经过测试，能成功计算文件与文件夹的Hash值并对key-value进行存储。
@wxc1217 @RitaPeng0101

ZhangNan13 committed:
方法名称修改

依照大家的建议，在原来的基础上修改了方法名，深度遍历并存储Hash值的方法名称修改为Show_KVstore，求文件夹Hash值的方法名称修改为Treehash。
@wxc1217 @RitaPeng0101

ZhangNan13 committed:
Hash修改

1.对文件存储及读取方法进行了修正，将路径变量添加为输入值，增加了文件存储及读取方法的可复用性。
2.新增了initsetting方法，用于创建存储git内容的文件夹，解决文件存储时无法找到路径的问题。

ZhangNan13 committed:
Hash修改与测试

1.对文件存储及读取方法进行了修正，将路径变量添加为输入值，增加了文件存储及读取方法的可复用性。
2.新增了initsetting方法，用于创建存储git内容的文件夹，解决文件存储时无法找到路径的问题。
3.新增了Hash类的测试。

ZhangNan13 committed:
Hash类显示修改

修改了Hash类的显示内容，对文件夹进行遍历存储时，只会显示未存储或已更新的文件的Hash值。
@wxc1217 @RitaPeng0101

wxc1217 commented on Hash.java:
```diff
@@ -0,0 +1,196 @@
+    public static String objectpath = "E:\\MyGit\\object";
+
+    /* 创建程序所需要的储存空间 */
+    public static void initsetting()
```
我觉得这里应该再加一个src文件夹，新增的文件放在里面。如果没有这个文件夹，新增文件直接在mygit仓库里面与object和branch文件夹并列的时候，后续进行commit之后，会直接输出所有文件的hash列表，（branch和object也会被求）。指定src文件路径，commit的时候只会求出新增文件的hash列表。

wxc1217 commented on Hash.java:
```diff
@@ -0,0 +1,196 @@
+    }
+
+    /* 对当前文件夹进行遍历，储存文件及文件夹的key-value值 */
+    public static String Show_KVstore(String path) throws IOException {
```
这里可以把展示和存储分开。commit的时候如果新增文件很多，则最终返回的hash列表会非常长，比较冗余。进行文件查找的时候，才需要hash列表的展示。

ZhangNan13 committed:
Hash类修改

1.初始化生成了src文件夹，用于commit部分Hash值计算。
2.修改了Hash类中key-value存储的显示问题，由于用户不需要了解文件与文件夹Hash值，不再对其进行显示。
@wxc1217 @RitaPeng0101
